ccvm_simulators/solvers/ccvm_solver.py

```diff
@@ -142,3 +157,123 @@ def _method_selector(self, problem_category):
                 "The given instance is not a valid problem category."
                 f" Given category: {problem_category}"
             )
+
+    def _validate_dataframe_columns(self, dataframe):
+        """Validates that the given dataframe contains the required columns when
+        calculating optics machine energy on DL-CCVM and MF-CCVM solvers.
+
+        Args:
+            dataframe (DataFrame): The dataframe to validate.
+
+        Raises:
+            ValueError: If the given dataframe is missing any of the required columns.
+        """
+        required_columns = ["pp_time", "iterations"]
+
+        missing_columns = [
+            col for col in required_columns if col not in dataframe.columns
+        ]
+
+        if missing_columns:
+            raise ValueError(
+                f"The given dataframe is missing the following columns: {missing_columns}"
+            )
+
+    def cpu_energy_max(self, machine_parameters: dict = None):
+        """The wrapper function of calculating the maximum power consumption of the
```

Comment:
The wrapper function of calculating the maximum "energy" consumption of the ...

Reply:
Fixed that in the other PR. https://github.com/1QB-Information-Technologies/ccvm/pull/168